File: Backend/models/product.js
import mongoose from "mongoose";
const { Schema, model } = mongoose;

const product = new Schema(
  {
    //poduct id
    product_id: {
      type: mongoose.Schema.Types.ObjectId,
    },

    user_id: String,
    category: {
      type: String,
      required: true,
    },

    name: {
      type: String,
      required: true,
    },

    description: {
      type: String,
      required: true,
    },
    price: {
      type: Number,
      required: true,
    },
    image: {
      type: Array,
      required: false,
    },

    quantity: {
      type: Number,
      default: 1,
    },
    added_date: { type: Number, default: new Date().getTime() },
    for_bid: { type: Boolean, default: false },
    bid_end_date: { type: Number },
    bade: { type: Boolean, default: false },
    sold: { type: Boolean, default: false },
    username: { type: String },
    userPhoneNumber: { type: String },
    userProfilePicture: { type: String },
    selling: { type: Boolean, default: false }, 


  },
  {
    timestamps: true,
  }
);

export default model("Product", product);

File: app/src/main/java/com/tradeasy/domain/product/entity/Product.kt
package com.tradeasy.domain.product.entity

import com.google.gson.annotations.SerializedName


data class Product(

    @SerializedName("user_id") val userId: String?,
    @SerializedName("category") val category: String?,
    @SerializedName("name") val name: String?,
    @SerializedName("description") val description: String?,
    @SerializedName("price") val price: Float?,
    @SerializedName("image") val image: List<String>?,
    @SerializedName("quantity") val quantity: Int?,
    @SerializedName("added_date") val addedDate: Number?,
    @SerializedName("for_bid") val forBid: Boolean?,
    @SerializedName("bid_end_date") val bidEndDate: Number?,
    @SerializedName("bade") val bade: Boolean?,
    @SerializedName("sold") val sold: Boolean?,
    @SerializedName("username") val username: String?,
    @SerializedName("userPhoneNumber") val userPhoneNumber: String?,
    @SerializedName("userProfilePicture") val userProfilePicture: String?,
    @SerializedName("selling") val selling: Boolean?,
    @SerializedName("_id") val productId: String?,


    )
// data class constructor
{
    constructor() : this(
        "",
        "",
        "",
        "",
        0.0f,
        listOf(),
        0,
        0,
        true,
        System.currentTimeMillis() + 860000,
        false,
        false,
        "",
        "",
        "",
        false,
        ""
    )
}






